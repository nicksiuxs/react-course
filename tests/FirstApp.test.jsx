import { render } from "@testing-library/react";
import FirstApp from "../src/FisrtApp";

describe("Test in FirstApp.test.jsx", () => {
  test("Should match the component with the snapshot", () => {
    const title = "Soy un titulo";

    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test("Should show the title sent by props in a h1", () => {
    const title = "Soy un titulo";

    const { container, getByText } = render(<FirstApp title={title} />);

    expect(getByText(title)).toBeTruthy();

    // This is not the best way to test this
    const h1 = container.querySelector("h1");
    expect(h1.innerHTML).toContain(title);
  });
});
