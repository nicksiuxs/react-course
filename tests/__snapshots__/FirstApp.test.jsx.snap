// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in FirstApp.test.jsx Should match the component with the snapshot 1`] = `
<div>
  <h1>
    Soy un titulo
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay nombre
  </p>
</div>
`;
